### DISCUSSION TITLE
Using `sxiv` to open images

### DISCUSSION THREAD

**Discussion by krackout** [gokcehan/lf#808] [Q&A] [answered]
I use **sxiv** to open images. In order to be able to do the same through **lf** I added this line to `lfrc`:

```
case $(file --mime-type $f -b) in
    image/*) setsid -f sxiv $f;;
```

Needed `setsid -f` to fork the process, so that **lf** is not locked.

It works, it opens images. But, I cannot use a feature of **sxiv**; hitting Enter key shows a preview of all images in the directory. If I open an image from **lf** hitting Enter in **sxiv** shows a preview of the opened image only, as if it's isolated. On the contrary, through **ranger** + **sxiv** I can preview all images in directory.

In `lfrc`, I tried `$fx` `"$f"` `"$fx"` instead of `sxiv $f`, no difference.  
Any ideas how to achieve?

Also, what's the difference between `$f`, `$fx`, `$fs`?

**Comment by xezwary** [comment, marked as answer]
Looks like Ranger uses a script that sends all files in the directory to sxiv
https://github.com/nsxiv/nsxiv-extra/tree/master/scripts/nsxiv-rifle


Here's the code I use in lfrc:
```
*.jpg|*.png|*.jpeg|*.JPG|*.JPEG|*.PNG|*.webp|*.ico|*.tga|*.gif)
    [[ $(echo "$fx"|wc -l) == 1 ]]&&imagelist=$(find . -maxdepth 1 -regextype posix-extended -iregex '.*(jpg|png|jpeg|webp|ico|tga|gif)'|sort -fV)&&for i in $imagelist;do echo $i;done|nsxiv -g 2560x1440 -s f -a -n $(echo "${imagelist[*]}"|grep --line-number --fixed-strings -- "./$(basename $f)"|cut -f1 -d:) -- -||for i in $fx;do echo $i;done|sort -fV|nsxiv -g 2560x1440 -s f -a -- - &
;;
```
Edit: Should probably add `set ifs "\n"` in lfrc

**Reply by krackout** [reply]
Great, I wouldn't image the mechanism of previewing works through through **Ranger/rifle**, thought that it was **sxiv** side implementation. I haven't yet made your setup to work on my lfrc, but I'm on the right track now. Have you `set shell bash`?

I changed it to **bash**, else `[[` produced errors; also I want to combine extensions with MIME types.
```
cmd open ${{
    test -L $f && f=$(readlink -f $f)

    case $f in
        *.jpg|*.png|*.jpeg|*.JPG|*.JPEG|*.PNG|*.webp|*.ico|*.tga|*.gif|*.GIF|*.bmp|*.BMP)
		[[ $(echo "$fx"|wc -l) == 1 ]]&&imagelist=$(find . -maxdepth 1 -regextype posix-extended -iregex '.*(jpg|png|jpeg|webp|ico|tga|gif|bmp)'|sort -fV)&&for i in $imagelist;do echo $i;done|sxiv $(echo "${imagelist[*]}"|grep --line-number --fixed-strings -- "./$(basename $f)"|cut -f1 -d:) -- -||for i in $fx;do echo $i;done|sort -fV|sxiv -- - &
	;;
    esac

    case $(file --mime-type $f -b) in
        text/plain) tmux split-window -h "less -MS '$fx'";;
        */pdf) setsid -f mupdf $fx;;
        *) for f in $fx; do setsid xdg-open $f > /dev/null 2> /dev/null & done;;
    esac
}}
```
Here is my non-working setup. It opens the first pic file twice, two **sxiv** windows. Any ideas how to make it work?
___
**Update:**
Apparently, **sxiv** opens twice, once by extension (first `case`) the second by default mime type (second `case`). So I located ranger's sxiv file and did this:
```
cmd open ${{
    test -L $f && f=$(readlink -f $f)

    case $(file --mime-type $f -b) in
        text/plain) tmux split-window -h "less -MS '$fx'";;
        image/*) setsid -f /usr/share/doc/ranger/examples/rifle_sxiv.sh $f;;
        */pdf) setsid -f mupdf $fx;;
        *) for f in $fx; do setsid xdg-open $f > /dev/null 2> /dev/null & done;;
    esac
}}
```
It seems that now I've got the desired result.

**Reply by xezwary** [reply]
I'm using zsh, looks like my code doesn't work properly with bash, glad you got it working though

Oh hey, just add `IFS=$(echo -en "\n\b")` and it works with bash
```
*.jpg|*.png|*.jpeg|*.JPG|*.JPEG|*.PNG|*.webp|*.ico|*.tga|*.gif)
    IFS=$(echo -en "\n\b")&&[[ $(echo "$fx"|wc -l) == 1 ]]&&imagelist=$(find . -maxdepth 1 -regextype posix-extended -iregex '.*(jpg|png|jpeg|webp|ico|tga|gif)'|sort -fV)&&for i in $imagelist;do echo $i;done|nsxiv -g 2560x1440 -s f -a -n $(echo "${imagelist[*]}"|grep --line-number --fixed-strings -- "./$(basename $f)"|cut -f1 -d:) -- -||for i in $fx;do echo $i;done|sort -fV|nsxiv -g 2560x1440 -s f -a -- - &
;;
```
